lib/services/dashd.js

```diff
@@ -802,12 +802,26 @@ Dash.prototype._initZmqSubSocket = function(node, zmqUrl) {
   var self = this;
   node.zmqSubSocket = zmq.socket('sub');
 
+  log.info(`ZMQ opening socket to '${zmqUrl}'`);
+
+  // workaround for https://github.com/zeromq/zeromq.js/issues/574
+  var tcpTimeout = 5 * 1000;
+  var timeout = setTimeout(function () {
+    // neither accepting nor rejecting - either invalid or firewalled (DROP)
+    log.error(`ZMQ address '${zmqUrl}' cannot be reached`);
+    process.exit(1);
```

Comment:
Will spawned process will be killed as well?

Reply:
Yes. You have to pass `{ detached: true }` to make a spawned process run independently from its parent:

https://nodejs.org/api/child_process.html#optionsdetached

(default behavior is that children die when the parent dies)